src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/OptionDescriptorExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc.OptionDescriptors;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Extension methods for collections of option descriptors.
+    /// </summary>
+    public static class OptionDescriptorExtensions
+    {
+        /// <summary>
+        /// Returns the only instance of <typeparamref name="TInstance"/> from a sequence of option descriptors.
```

Comment:
@tonysneed @yishaigalatzer I find this method's behaviour surprising:
- expected a method with this name to `null` if none are found. put another way, why does `InstanceOfOrDefault()` exist?
- didn't expect this method (or `InstanceOfOrDefault()` if we need it) to `throw` when multiple matches are found. glad the behaviour is described but why aren't `InstanceOf()` and `InstancesOf()` more like for example `GetCustomAttribute()` and `GetCustomAttributes()`?
 - if we do want this behaviour, why defer to `Single()` to `throw` the `InvalidOperationException` rather than craft a specific message in MVC?

- expected this method to return any subclass of `TInstance` i.e. behave more like `OfType<TInstance>().FirstOrDefault()` (or `...Single()` if we stick w/ current behaviour on the other two points). but we're having that conversation elsewhere :smiley_cat: